### PR TITLE
Add metric names endpoint

### PR DESCRIPTION
**Pull request by rbrndt**

### CODE REVIEW

File: monascaclient/metrics.go [hunk 1]
```diff
@@ -87,7 +92,7 @@ func (c *Client) getDimensionQuery(path string, dimensionQuery interface{}) ([]s
 	response := new(models.DimensionValueResponse)
 	err := c.callMonascaGet(metricsBasePath+path, "", dimensionQuery, response)
 	if err != nil {
-		return nil, err
```
Comment (mhoppal) [inline monascaclient/metrics.go:90]: why change to empty structs when there is an error?

File: monascaclient/metrics.go [hunk 2]
```diff
@@ -98,6 +103,20 @@ func (c *Client) getDimensionQuery(path string, dimensionQuery interface{}) ([]s
 	return results, nil
 }
 
+func (c *Client) GetMetricNames(metricQuery *models.MetricNameQuery) ([]string, error) {
+	response := new(models.MetricNameResponse)
+	err := c.callMonascaGet(metricsBasePath+"/names", "", metricQuery, response)
```
Comment (mhoppal) [inline monascaclient/metrics.go:108]: check spacing

File: monascaclient/metrics.go (hunk 1 above)
Comment (rbrndt) [reply monascaclient/metrics.go:90]: nil for a struct will cause a panic

File: monascaclient/metrics.go (hunk 2 above)
Comment (rbrndt) [reply monascaclient/metrics.go:108]: gofmt chose this spacing, so I left it alone

**Pull request merged**